blocks/main/latest-news/mascot/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { memo, useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { useInView } from 'react-intersection-observer';
+import { AnimationItem, LottiePlayer } from 'lottie-web/build/player/lottie_svg';
+
+import styles from './mascot.module.css';
+
+type MascotProps = {
+    className?: string;
+};
+
+const ANIMATION_INITIAL_DELAY = 1500 as const;
+const ANIMATION_AFTER_DELAY = 7000 as const;
```

Comment:
"as const" - could you please explain why we need it here? I did't expect to see it with const declarations.

Reply:
Thanks, it's a goooooooood question! :)

Mostly it has some useful cases for *complex type* like `Record` or `Array`.
And I know just single so synthetic case for *primitive type*:
```javascript
const OTHER_CONST = 123;

const TIMEOUT_1 = 1000 as const;
const TIMEOUT_2 = 2000 as const;

type TIMEOUTS = typeof TIMEOUT_1 | typeof TIMEOUT_2; // i've invented `enum`!

function print(n: number) {
  console.log(n);
}

function timeout(n: TIMEOUTS) {
  console.log(n);
}

print(OTHER_CONST);
print(TIMEOUT_1);
print(TIMEOUT_2);

timeout(OTHER_CONST); // oops...
timeout(TIMEOUT_1);
timeout(TIMEOUT_2);
```
But of course, it isn't there.
I just use `as const` for all `ANY_CONST` without mind cause i don't know any side effects.
I'll drop it if it's clear.

Reply:
Ah, I see now. Didn't now about "const assertions" in TS. [For the reference](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions).